server/src/main/kotlin/fi/oph/kitu/logging/LogEventService.kt

```diff
@@ -0,0 +1,37 @@
+package fi.oph.kitu.logging
+
+import fi.oph.kitu.ExternalSystem
+import org.slf4j.spi.LoggingEventBuilder
+import org.springframework.stereotype.Service
+import java.net.http.HttpResponse
+
+data class ResponseData(
+    val externalSystem: ExternalSystem,
+    val headers: Map<String, List<String>>,
+    val body: String,
+)
+
+// Singleton instance in order to access the eventDataContext
+@Service
+class LogEventService {
+    var event: LoggingEventBuilder? = null
```

Comment:
Ovatko `@Service`-instanssit per-thread vai onko niitä vain yksi? Tällainen services scopessa oleva tila haiskahtaa hieman thread-safety ongelmilta; mitä jos useampi thread asettaa `logEventService.event = logger.atInfo()` lähes samanaikaisesti?

Reply:
@Service pitäis olla singleton. Sen lifecycle on mulle vähä epämääräinen, mut sama instanssi pitäis olla elossa koko requestin ajan.

Reply:
En ole huolissani niinkään onko `@Service` olemassa koko requestin ajan, vaan voiko useamman requestin käsittely käyttää samaa service instanssia samanaikaisesti. Tarkoittaako "singleton" tässä yhtä globaalia instanssia, jota kaikki threadit käyttävät, vai onko jokaiselle threadille oma instanssinsa?

Mikäli meillä on vain yksi `LogEventService` -instanssi ja mikäli pyyntöjä käsitellään rinnakkain useammassa threadissa, useampi thread voi asettaa `logEventService.event` lähes samanaikaisesti, jolloin `.addResponses` saattaa kirjoittaa väärän pyynnön lokieventtiin.

Reply:
Ratkaistu asia poistamalla service-viritelmä-commitit.